src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php

```diff
@@ -73,14 +73,14 @@ public function parse($controller)
             throw new \InvalidArgumentException($message, 0, $e);
         }
 
-        foreach ($allBundles as $b) {
-            $try = $b->getNamespace().'\\Controller\\'.$controller.'Controller';
-            if (class_exists($try)) {
-                return $try.'::'.$action.'Action';
+        foreach ($allBundles as $bundle) {
```

Comment:
this is not good. It overwrite the `$bundle` variable defined earlier, while it is still necesaary after to build the error message